### ISSUE TITLE
Scout Generating CSS in input and output folders

### ISSUE THREAD

**Issue by dshastry** [scout-app/scout-app#343]
I've been using scout with much success on a few projects. Been working great until yesterday when it started spitting a weird bug. The app seems to generate random css in the input directory. It is a different SCSS file that's generated each time I save.

Here's my input/output directories setup in different folders
![image](https://user-images.githubusercontent.com/4194561/28132872-c0b92ee0-6703-11e7-90b5-4ffebf34cf13.png)

This is the error I get when I save an SCSS file in my input folder: 
![image](https://user-images.githubusercontent.com/4194561/28132899-d7a3c23c-6703-11e7-95dd-f0995cd78543.png)

And here's what I'm seeing in my input folder:
![image](https://user-images.githubusercontent.com/4194561/28132947-fd361356-6703-11e7-88dd-80f5698c72dd.png)

My output folder is generating just fine with the required CSS but I'm getting phantom CSS files in the input that then messes up future SCSS generation. I have to manually delete the CSS files and then Scout will run again without error. Then when I resave it'll generate the CSS files in the input again and I have to manually delete again.

This behaviour does not happen if I quit the app and restart or I stop and start the projects. Anyone else face this?

**Comment by TheJaredWilcurt** [comment]
The only time Scout-App writes a file to the drive is when you either save settings, or when Node-Sass has returned a string of CSS or Mapping data to be saved to the output folder. The the path for saving CSS/Mapping must always begin with whatever is in the "output" setting for your project.

The only way this issue could occur is if your input and output settings are confusing Scout-App.

To be able to debug this, I need to see the settings for your project. Go to `View > Dev Tools` in Scout-App and then run the following code in the console:

```js
require('path').join(require('nw.gui').App.dataPath,'scout-settings.json');
```

That will give you the location of the scout-settings file on your computer. Upload that as a response here (or to something like [Gist](https://gist.github.com) and link to it). Then I'll be able to do more thorough tests and try to reproduce the issue. File paths can be tricky some times.

For the time being, you can change your input code to look for specific files by including their file extension.
```scss
@import "somefile.scss";
```
instead of
```scss
@import "somefile";
```

**Comment by dshastry** [comment]
Adding the extension seemed to fix it temporarily @TheJaredWilcurt! Also realized that even though I'm using brackets with Sass disabled as my code editor there's still a conflict. Switching to sublime text or some other editor doesn't cause this issue.

**Comment by TheJaredWilcurt** [comment]
@dshastry 
can you still copy/paste the input and output folders into here so I can verify that this isn't an issue with the app's logic.

**Comment by dshastry** [comment]
Sure. Here you go: `{
    "projects": [
        {
            "projectID": "sa1498593231153",
            "projectFolder": "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS",
            "projectName": "CSS",
            "imageFolder": "",
            "projectIcon": "_img/logo_128.png",
            "inputFolder": "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/SCSS",
            "outputFolder": "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/CSS",
            "environment": "production",
            "outputStyle": "compressed",
            "indicator": "stop",
            "watcher": {
                "domain": null,
                "_events": {},
                "_watched": {
                    "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS": {
                        "_items": {
                            "SCSS": true
                        }
                    },
                    "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/SCSS": {
                        "_items": {
                            "global": true,
                            "modules": true,
                            "partials": true,
                            "plugins": true,
                            "Icon\r": true,
                            "base.scss": true
                        }
                    },
                    "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/SCSS/global": {
                        "_items": {
                            "Icon\r": true,
                            "colors.scss": true,
                            "fonts.scss": true,
                            "global-styles.scss": true
                        }
                    },
                    "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/SCSS/modules": {
                        "_items": {
                            "Icon\r": true,
                            "docgen.scss": true,
                            "integrations.scss": true,
                            "sms.scss": true
                        }
                    },
                    "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/SCSS/partials": {
                        "_items": {
                            "Icon\r": true,
                            "bs-equal-height-rows.scss": true,
                            "forms.scss": true,
                            "normalize.scss": true
                        }
                    },
                    "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/SCSS/plugins": {
                        "_items": {
                            "dependencies": true,
                            "Icon\r": true,
                            "alertify.scss": true,
                            "bs-duallistbox.scss": true,
                            "bs-modal.scss": true,
                            "bs-toggle.scss": true,
                            "editable.scss": true,
                            "jq-tabs.scss": true,
                            "niceselect.scss": true,
                            "split-panes.scss": true,
                            "tables.scss": true,
                            "bs-modal.css": true,
                            "bs-modal.css.map": true
                        }
                    },
                    "/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS/SCSS/plugins/dependencies": {
                        "_items": {
                            "Icon\r": true,
                            "alertify.min.scss": true,
                            "bootstrap-duallistbox.scss": true,
                            "bootstrap-toggle.min.scss": true,
                            "bootstrap.min.scss": true,
                            "x-editable.min.scss": true
                        }
                    }
                },
                "_closers": {},
                "_ignoredPaths": {},
                "closed": false,
                "_throttled": {
                    "readdir": {},
                    "add": {},
                    "watch": {},
                    "remove": {},
                    "change": {}
                },
                "_symlinkPaths": {},
                "enableBinaryInterval": true,
                "options": {
                    "ignored": [
                        {}
                    ],
                    "persistent": true,
                    "ignoreInitial": false,
                    "ignorePermissionErrors": false,
                    "interval": 100,
                    "binaryInterval": 300,
                    "useFsEvents": false,
                    "usePolling": true,
                    "atomic": false,
                    "followSymlinks": true,
                    "awaitWriteFinish": false
                },
                "_readyCount": 44,
                "_readyEmitted": true
            },
            "hasSassFiles": true
        },
        {
            "projectID": "sa1499720188110",
            "projectFolder": "/Users/davidshastry/Desktop/client-portal/assets/scss",
            "projectName": "scss",
            "imageFolder": "",
            "projectIcon": "_img/logo_128.png",
            "inputFolder": "/Users/davidshastry/Desktop/client-portal/assets/scss",
            "outputFolder": "/Users/davidshastry/Desktop/client-portal/assets/css",
            "environment": "production",
            "outputStyle": "compressed",
            "indicator": "stop",
            "watcher": {
                "domain": null,
                "_events": {},
                "_watched": {
                    "/Users/davidshastry/Desktop/client-portal/assets": {
                        "_items": {
                            "scss": true
                        }
                    },
                    "/Users/davidshastry/Desktop/client-portal/assets/scss": {
                        "_items": {
                            "global": true,
                            "plugins": true,
                            "partials": true,
                            "client-portal.scss": true
                        }
                    },
                    "/Users/davidshastry/Desktop/client-portal/assets/scss/global": {
                        "_items": {
                            "colors.scss": true
                        }
                    },
                    "/Users/davidshastry/Desktop/client-portal/assets/scss/plugins": {
                        "_items": {
                            "dependencies": true,
                            "cards.scss": true,
                            "flexbox.scss": true,
                            "header.scss": true,
                            "login-form.scss": true,
                            "modal.scss": true
                        }
                    },
                    "/Users/davidshastry/Desktop/client-portal/assets/scss/partials": {
                        "_items": {
                            "animation.scss": true,
                            "fonts.scss": true
                        }
                    },
                    "/Users/davidshastry/Desktop/client-portal/assets/scss/plugins/dependencies": {
                        "_items": {
                            "bootstrap.min.scss": true
                        }
                    }
                },
                "_closers": {},
                "_ignoredPaths": {},
                "closed": false,
                "_throttled": {
                    "readdir": {},
                    "add": {}
                },
                "_symlinkPaths": {},
                "enableBinaryInterval": true,
                "options": {
                    "ignored": [
                        {}
                    ],
                    "persistent": true,
                    "ignoreInitial": false,
                    "ignorePermissionErrors": false,
                    "interval": 100,
                    "binaryInterval": 300,
                    "useFsEvents": false,
                    "usePolling": true,
                    "atomic": false,
                    "followSymlinks": true,
                    "awaitWriteFinish": false
                },
                "_readyCount": 15,
                "_readyEmitted": true
            },
            "hasSassFiles": true
        }
    ],
    "versions": {
        "nodeSass": "3.13.1",
        "libSass": "3.3.6",
        "scout": "2.12.12",
        "chokidar": "1.6.1"
    },
    "globalSettings": {
        "cultureCode": "en",
        "alertDesktop": true,
        "alertInApp": true,
        "alertSound": true,
        "messageDesktop": true,
        "messageInApp": true,
        "messageSound": true
    }
}
`

**Comment by TheJaredWilcurt** [comment]
@dshastry 

I think what might be related to the issue is that your project folders are set to:

* `/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal/CSS`
* `/Users/davidshastry/Desktop/client-portal/assets/scss`

when they should be set to:

* `/Users/davidshastry/Google Drive/Debt Pay Pro Design/DESIGN/Debt Pay Pro Portal`
* `/Users/davidshastry/Desktop/client-portal`

You can do this most easily, by just dragging the `client-portal` folder from the desktop into Scout-App, or the `Debt Pay Pro Portal` folder into Scout-App. Then verify the input and output folders are correct.

What's more interesting is your settings show you're in `production` and not `development`, which means Scout-App can't produce `.map` files.

Scout-App only looks for `.scss` and `.sass` files, and will ignore all other files in the input folders. It also generates `file.map` files not `file.map.css`. I'm wondering if you have another program processing things too.

**Comment by dshastry** [comment]
@TheJaredWilcurt, I use Scout for multiple projects and those are two separate projects. Here's how I organize the dev/production code for my projects usually: 
![image](https://user-images.githubusercontent.com/4194561/28280005-f5586aca-6ae7-11e7-9533-6a6d56bad3b8.png)

All the SCSS files are in the SCSS input folder and the compiled goes into CSS. I was pushing to production so yes, I had it set to prod/minified.

What I realized the issue was was that Brackets as the text editor was messing up on the node server side. Even though I disabled SCSS compiling in brackets that didn't propagate to the Node backend and it was still running jobs. This was tripping up Scout because it would find multiple files with the same name in the input folder and then stop processing (as expected).

The fix for me was to completely uninstall Brackets and Node, get the newest version of both, run a clean install and then disable SCSS compiling once again. With all that taken care of, Scout runs like a breeze as expected. So yes, looks like bugs in other programs :\
